Program.cs

```diff
@@ -9,19 +9,24 @@ internal class Program {
          if (isStrong) Console.WriteLine ("Strong password.");
          else Console.WriteLine ("Password is not strong.\n" + errorMsg);
       }
+      static bool isLower (char c) {
+         return (c > 'a' && c < 'z');
+      }
+      static bool isUpper (char c) {
+         return (c > 'A' && c < 'Z');
+      }
+      static bool isDigit (char c) {
+         return (c > '0' && c < '9');
+      }
       static (bool isStrong, string errorMsg) checkPassword (string pass) {
          StringBuilder err = new ();
+         string spl = "!@#$%^&*()-+";
          bool lower = false, upper = false, digit = false, splChar = false;
          foreach (char c in pass) {
-            if (char.IsLower (c))
-               lower = true;
-            if (char.IsUpper (c))
-               upper = true;
-            if (char.IsDigit (c))
-               digit = true;
-            string spl = "!@#$%^&*()-+";
-            if (spl.Contains (c))
-               splChar = true;
+            if (isLower (c)) lower = true;
+            else if (isUpper (c)) upper = true;
+            else if (isDigit (c)) digit = true;
+            else if (spl.Contains (c)) splChar = true;

```

Comment:
Unsupported special characters should not be allowed. [In this example, tilde (~) or quotes (") should not be allowed]

Reply:
Changes done in such a way  that if only valid characters are entered, the input is checked for strong password.